Build System: defconfig: Base defconfig not used on fully qualified oot board

### Describe the bug

When using a fully qualified board (I'm using a test `native_sim/native/64/two`) my interpretation of the documentation [here](https://docs.zephyrproject.org/latest/hardware/porting/board_porting.html#write-kconfig-files) is that the the defconfig files should be matched with `native_sim_defconfig`, as well as each file getting more specific with each qualifier
(e.g. `native_sim_defconfig`, `native_sim_native_defconfig`, `native_sim_native_64_defconfig`, `native_sim_native_64_two_defconfig`)

> If both the common `plank_defconfig` file and one or more board qualifiers specific `plank_<qualifiers>_defconfig` files exist, then all matching files will be used. This allows you to place configuration which is common for all board SoCs, CPU clusters, and board variants in the base `plank_defconfig` and only place the adjustments specific for a given SoC or board variant in the `plank_<qualifiers>_defconfig`.

Depending on your interpretation of the documentation, the you may have a different opinion on which configs with "intermediate" qualifiers should be included... However, I would consistently expect at least `native_sim_defconfig` and `native_sim_native_64_two_defconfig` to be used regardless of interpretation of docs **and it is not**. 

Right now only `native_sim_native_64_two_defconfig` is used. I have reproduced this in a ztest test case available on a public fork of the Zephyr repo


### Regression

- [ ] This is a regression.

### Steps to reproduce

Steps to reproduce (on a Linux machine):
1. Check out the branch `luskey/qualifier-maybe-bug` from https://github.com/LuskeyNoah/zephyr/ (or simply cherry pick [this commit](https://github.com/zephyrproject-rtos/zephyr/commit/55c4ed63f984af953890a21452d50b55822fd3dc))
2. Navigate to `tests/cmake/hwm/board_extend` and build for `native_sim/native/64/two` (`west build -b native_sim/native/64/two`)
3. Run the zephyr application and observe the results

### Relevant log output

```shell
(.venv) ➜  board_extend git:(luskey/qualifier-maybe-bug) west build -t run
-- west build: running target run
[1/10] Generating include/generated/zephyr/version.h
-- Zephyr version: 4.2.99 (/home/luskey/zephyr-workdir/zephyr), build: v4.2.0-6916-g55c4ed63f984
[7/10] Linking C executable zephyr/zephyr.elf
Generating files from /home/luskey/zephyr-workdir/zephyr/tests/cmake/hwm/board_extend/build/zephyr/zephyr.elf for board: native_sim
[9/10] cd /home/luskey/zephyr-workdir/zephyr/tests/cmake/hwm/board_extend/build && /home/luskey/zephyr-workdir/zephyr/tests/cmake/hwm/board_extend/build/zephyr/zephyr.exe
*** Booting Zephyr OS build v4.2.0-6916-g55c4ed63f984 ***
Running TESTSUITE soc_board_extend
===================================================================
START - test_native_sim_extend
 PASS - test_native_sim_extend in 0.000 seconds
===================================================================
START - test_qualifier_matching

    Expectation failed at WEST_TOPDIR/zephyr/tests/cmake/hwm/board_extend/src/main.c:108: soc_board_extend_test_qualifier_matching: IS_ENABLED(CONFIG_SET_BY_NATIVE_SIM) is false


    Expectation failed at WEST_TOPDIR/zephyr/tests/cmake/hwm/board_extend/src/main.c:111: soc_board_extend_test_qualifier_matching: IS_ENABLED(CONFIG_SET_BY_NATIVE_SIM_NATIVE) is false


    Expectation failed at WEST_TOPDIR/zephyr/tests/cmake/hwm/board_extend/src/main.c:114: soc_board_extend_test_qualifier_matching: IS_ENABLED(CONFIG_SET_BY_NATIVE_SIM_NATIVE_64) is false

 FAIL - test_qualifier_matching in 0.000 seconds
===================================================================
TESTSUITE soc_board_extend failed.

------ TESTSUITE SUMMARY START ------

SUITE FAIL -  50.00% [soc_board_extend]: pass = 1, fail = 1, skip = 0, total = 2 duration = 0.000 seconds
 - PASS - [soc_board_extend.test_native_sim_extend] duration = 0.000 seconds
 - FAIL - [soc_board_extend.test_qualifier_matching] duration = 0.000 seconds

------ TESTSUITE SUMMARY END ------

===================================================================
PROJECT EXECUTION FAILED
FAILED: zephyr/CMakeFiles/run_native /home/luskey/zephyr-workdir/zephyr/tests/cmake/hwm/board_extend/build/zephyr/CMakeFiles/run_native 
cd /home/luskey/zephyr-workdir/zephyr/tests/cmake/hwm/board_extend/build && /home/luskey/zephyr-workdir/zephyr/tests/cmake/hwm/board_extend/build/zephyr/zephyr.exe
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /usr/bin/cmake --build /home/luskey/zephyr-workdir/zephyr/tests/cmake/hwm/board_extend/build --target run
```

### Impact

Functional Limitation – Some features not working as expected, but system usable.

### Environment

- OS: Linux
- Toolchain: 0.17.4, x86_64-zephyr-elf
- Commit (fork): 55c4ed63f984af953890a21452d50b55822fd3dc

### Additional Context

_No response_

--- a/doc/hardware/porting/board_porting.rst
+++ b/doc/hardware/porting/board_porting.rst
@@ -367,12 +367,11 @@ If multiple boards are placed in the same board folder, then the file
 Write your devicetree
 *********************
 
-The devicetree file :file:`boards/<vendor>/plank/plank.dts` or
-:file:`boards/<vendor>/plank/plank_<qualifiers>.dts` describes your board
+The devicetree file :file:`boards/<vendor>/plank/plank_<qualifiers>.dts` describes your board
 hardware in the Devicetree Source (DTS) format (as usual, change ``plank`` to
 your board's name). If you're new to devicetree, see :ref:`devicetree-intro`.
 
-In general, :file:`plank.dts` should look like this:
+In general, :file:`plank_<qualifiers>.dts` should look like this:
 
 .. code-block:: devicetree
 
@@ -422,16 +421,9 @@ In general, :file:`plank.dts` should look like this:
            status = "okay";
    };
 
-Only one ``.dts`` file will be used, and the most specific file which exists
-will be used.
-
-This means that if both :file:`plank.dts` and :file:`plank_soc1_foo.dts` exist,
-then when building for ``plank`` / ``plank/soc1``, then :file:`plank.dts` is
-used. When building for ``plank//foo`` / ``plank/soc1/foo`` the
-:file:`plank_soc1_foo.dts` is used.
-
-This allows board maintainers to write a base devicetree file for the board
-or write specific devicetree files for a given board's SoC or variant.
+In the case a board has only a single SoC, without any board variants then the dts file can be
+named :file:`<plank>.dts` instead, however this is not recommended due to the file silently be
+unused if a variant or other SoC is added to the board.
 
 If you're in a hurry, simple hardware can usually be supported by copy/paste
 followed by trial and error. If you want to understand details, you will need
@@ -516,7 +508,6 @@ files for a board named ``plank``:
    ├── Kconfig
    ├── Kconfig.plank
    ├── Kconfig.defconfig
-   ├── plank_defconfig
    └── plank_<qualifiers>_defconfig
 
 :file:`Kconfig.plank`
@@ -573,23 +564,28 @@ files for a board named ``plank``:
              default y
 
      if NETWORKING
+
      config SOC_ETHERNET_DRIVER
              default y
+
      endif # NETWORKING
 
      endif # BOARD_PLANK
 
-:file:`plank_defconfig` / :file:`plank_<qualifiers>_defconfig`
+:file:`plank_<qualifiers>_defconfig` (or :file:`plank_defconfig` in limited circumstances)
   A Kconfig fragment that is merged as-is into the final build directory
   :file:`.config` whenever an application is compiled for your board.
 
-  If both the common :file:`plank_defconfig` file and one or more board
-  qualifiers specific :file:`plank_<qualifiers>_defconfig` files exist, then
-  all matching files will be used.
-  This allows you to place configuration which is common for all board SoCs,
-  CPU clusters, and board variants in the base :file:`plank_defconfig` and only
-  place the adjustments specific for a given SoC or board variant in the
-  :file:`plank_<qualifiers>_defconfig`.
+  :file:`plank_defconfig` can only be used with boards that have no qualifiers, no variants and a
+  single SoC present, though this style of naming is not recommended due to samples/tests or
+  downstream usage breaking suddenly without warning if a new SoC or board variant/qualifier is
+  added to an board in upstream Zephyr.
+
+.. note::
+  Multiple files are not merged and there is no fallback mechanism for files, this means if there
+  is a board with 2 different SoCs and each one has 2 board variants, a :file:`plank_defconfig`
+  file would be wholly unused, for the first qualifier and variant
+  :file:`plank_<soc1>_<variant1>_defconfig` will be used, it will not include other file.
 
   The ``_defconfig`` should contain mandatory settings for your UART,
   console, etc. The results are architecture-specific, but typically look

